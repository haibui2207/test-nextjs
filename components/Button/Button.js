import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './button.styles';

const Button = ({
  size,
  color,
  disabled,
  children,
  dataTest,
  className,
  tag: Component,
  ...restProps
}) => {
  const classes = useStyles();

  return (
    <Component
      className={classNames(
        classes.root,
        size,
        color,
        disabled && 'disabled',
        className,
      )}
      disabled={disabled}
      data-test={dataTest || 'component-button-root'}
      {...restProps}
    >
      {children}
    </Component>
  );
};
Button.propTypes = {
  children: PropTypes.node.isRequired,
  tag: PropTypes.node,
  disabled: PropTypes.bool,
  dataTest: PropTypes.string,
  className: PropTypes.string,
  size: PropTypes.oneOf(['large', 'medium', 'small']),
  color: PropTypes.oneOf(['default', 'primary', 'secondary', 'tertiary']),
};
Button.defaultProps = {
  dataTest: '',
  className: '',
  tag: 'button',
  size: 'medium',
  disabled: false,
  color: 'primary',
};

export default memo(Button);
