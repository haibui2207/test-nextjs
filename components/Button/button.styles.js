import { createUseStyles } from 'react-jss';

import { rem } from '../../utils/jss';
import COLORS from '../../constants/COLORS';
import BREAKPOINTS from '../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    fontFamily: 'inherit',
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 500,
    lineHeight: rem(24),
    padding: [0, rem(16)],
    textAlign: 'center',
    borderRadius: rem(4),
    position: 'relative',
    transition: 'background-color .1s linear',
    whiteSpace: 'nowrap',
    userSelect: 'none',
    textDecoration: 'none',
    boxSizing: 'border-box',
    overflow: 'hidden',
    '&:focus': { outline: 'none' },
    '&:hover': { cursor: 'pointer' },
    '&.large': { height: rem(48), fontSize: rem(18) },
    '&.medium': { height: rem(40), fontSize: rem(16) },
    '&.small': { height: rem(32), fontSize: rem(14) },
    '&.default': {
      color: COLORS.textDarkGray,
      border: `1px solid ${COLORS.borderLight}`,
      backgroundColor: COLORS.bgWhite,
      '&.disabled': {
        cursor: 'not-allowed',
        color: COLORS.textGray,
        backgroundColor: COLORS.bgLightGray,
      },
    },
    '&.primary': {
      border: 'none',
      color: COLORS.textWhite,
      backgroundColor: COLORS.primary,
      '&.disabled': {
        cursor: 'not-allowed',
        color: COLORS.textGray,
        backgroundColor: COLORS.bgLightGray,
      },
    },
    '&.secondary': {
      color: COLORS.primary,
      border: `1px solid ${COLORS.primary}`,
      backgroundColor: COLORS.bgWhite,
      '&.disabled': {
        cursor: 'not-allowed',
        borderColor: COLORS.borderLight,
        color: COLORS.textDarkGray,
      },
    },
    '&.tertiary': {
      color: COLORS.textDarkBlack,
      border: 'none',
      backgroundColor: 'transparent',
      '&.disabled': { cursor: 'not-allowed', color: COLORS.textGray },
    },
  },

  '@media (hover: hover) and (pointer: fine)': {
    root: {
      '&.default:not(.disabled):hover': {
        color: COLORS.textWhite,
        backgroundColor: COLORS.primary,
      },
      '&.primary:not(.disabled):hover': {
        backgroundColor: COLORS.primaryDark,
      },
      '&.secondary:not(.disabled):hover': {
        backgroundColor: COLORS.primaryLight,
      },
      '&.tertiary:not(.disabled):hover': {
        color: COLORS.primary,
      },
    },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: {
      '&.large, &.medium, &.small': { fontSize: rem(14), height: rem(32) },
    },
  },
});
