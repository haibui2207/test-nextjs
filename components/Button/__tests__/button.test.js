import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

describe('Testing Button component', () => {
  it('should render match snapshot', () => {
    const wrapper = shallow(<Button>Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
