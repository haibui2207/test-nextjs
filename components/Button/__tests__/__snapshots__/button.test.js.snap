// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component should render match snapshot 1`] = `ShallowWrapper {}`;
